File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        res=[0]*26
        for x in range(len(s)):
            res[ord(s[x])-ord('a')]+=1
        for x in range(len(t)):
            res[ord(t[x])-ord('a')]-=1
        for x in res:
            if x!=0:
                return False
        return True

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prefix=1;
        int[]res=new int[nums.length];
        Arrays.fill(res,1);
        for(int i=0;i<nums.length;i++){   
            res[i]=prefix;
            prefix*=nums[i];
        }
        int postfix=1;
        for(int i=nums.length-1;i>=0;i--){
            res[i]*=postfix;
            postfix*=nums[i];
        }
        return res;
    }
}  



// 1 1 2 4 6 1
// 1 1 1 2 8 1
// 1 1 2 8
//  48  24 12  8

// 48 24 12 8